pwnlib/commandline/template.py

```diff
@@ -9,24 +9,54 @@
 parser = common.parser_commands.add_parser(
     'template',
     help = 'Generate an exploit template',
-    description = 'Generate an exploit template'
+    description = 'Generate an exploit template. If no arguments are given, '
+                    'the current directory is searched for an executable binary and ' 
+                    'libc. If only one binary is found, it is assumed to be the '
+                    'challenge binary.'
 )
 
 # change path to hardcoded one when building the documentation
 printable_data_path = "pwnlib/data" if 'sphinx' in sys.modules else pwnlib.data.path
 
-parser.add_argument('exe', nargs='?', help='Target binary')
+parser.add_argument('exe', nargs='?', help='Target binary. If not given, the current directory is searched for an executable binary.')
 parser.add_argument('--host', help='Remote host / SSH server')
 parser.add_argument('--port', help='Remote port / SSH port', type=int)
 parser.add_argument('--user', help='SSH Username')
 parser.add_argument('--pass', '--password', help='SSH Password', dest='password')
-parser.add_argument('--libc', help='Path to libc binary to use')
+parser.add_argument('--libc', help='Path to libc binary to use. If not given, the current directory is searched for a libc binary.')
 parser.add_argument('--path', help='Remote path of file on SSH server')
 parser.add_argument('--quiet', help='Less verbose template comments', action='store_true')
 parser.add_argument('--color', help='Print the output in color', choices=['never', 'always', 'auto'], default='auto')
 parser.add_argument('--template', help='Path to a custom template. Tries to use \'~/.config/pwntools/templates/pwnup.mako\', if it exists. '
                                    'Check \'%s\' for the default template shipped with pwntools.' % 
                                         os.path.join(printable_data_path, "templates", "pwnup.mako"))
+parser.add_argument('--no-auto', help='Do not automatically detect missing binaries', action='store_true')
```

Comment:
```suggestion
parser.add_argument('--no-auto', help='Do not automatically detect missing binaries', action='store_false', dest='auto')
```
I suppose we can keep less double negations in code